landoapi/hgexports.py

```diff
@@ -605,3 +616,144 @@ def run_diff_checks(self) -> list[str]:
                 issues.append(issue)
 
         return issues
+
+
+@dataclass
+class PushCheck:
+    """Provides an interface to implement push-wide checks."""
+
+    repo: Repo
+
+    def relevant(self) -> bool:
+        """Return `True` if the check is relevant for the given repo."""
+        raise NotImplementedError()
+
+    def next_diff(self, patch_helper: PatchHelper):
+        """Pass the next diff to be checked into the check and run."""
+        raise NotImplementedError()
+
+    def result(self) -> Optional[str]:
+        """Return the result of the check."""
+        raise NotImplementedError()
+
+
+@dataclass
+class BugReferencesCheck(PushCheck):
+    """Prevent commit messages referencing non-public bugs from try."""
+
+    bug_ids: set[int] = field(default_factory=set)
+    skip_check: bool = False
+
+    def relevant(self) -> bool:
+        if self.repo.tree == "try":
+            return True
+
+        return False
```

Comment:
I think putting the checks in `Repo` makes sense. We could enable/disable checks in the admin interface without needing a code change. I'm going to do that part in a follow-up because the most logical approach will require some changes to `DiffAssessor` and I want to keep this PR focused on finishing the last 2 checks.